internal/transport/transport_test.go

```diff
@@ -2090,7 +2090,7 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 			metaHeaderFrame: &http2.MetaHeadersFrame{
 				Fields: []hpack.HeaderField{
 					{Name: "content-type", Value: "application/grpc"},
-					{Name: "grpc-status", Value: "0"},
+					{Name: "grpc-status", Value: "14"},
```

Comment:
If there is a `grpc-status`, then there _should_ be a `grpc-message` to go along with it, or we should use the empty string like we do normally -- we should not be using the generated "unexpected HTTP status code" error message.  At least, that's what it _seems_ like it should be doing, though it's not mentioned in our docs.  Does that seem right to you, @ejona86 / @markdroth?

Maybe set this error code to a different number and change the `wantStatus` code, so it is clear that's the source of the error, not the 504->Unavailable conversion?

Reply:
I would say a grpc-message should go along with it from a "we don't hate ourselves" perspective. If there isn't a message, I wouldn't use an empty string; I'd leave the header missing. I wouldn't suggest a useless message like "UNAVAILABLE"; again, I'd prefer just not having it.

But that's a bit neither here nor there. In this case, there is a grpc-status, so you should use that and the grpc-message if it isn't present. I don't think this test is correct: the status shouldn't have a status description.

Reply:
> I'd leave the header missing

This is client-side.  If the header is missing, the client would use "" as the message string presented to the user.  (We have to give them _some_ string, since it's returned as a string value; we can't use `nil`.)

Reply:
Ah, then yeah, empty string seems appropriate for your API. Java's API lets the description (which we call grpc-message) be null. Makes sense that you wouldn't have such a distinction in Go.

Reply:
Ping @JNProtzman, I think this is the last issue: to not use the generated http error message along with the `grpc-status` header value in the status returned to the user.